libs/watt/src/lib/components/datetime-field/watt-datetime-field.component.ts

```diff
@@ -0,0 +1,204 @@
+
+  /** Converts date from outer FormControl to format of inner FormControl. */
+  protected modelToView = (value: Date | null, format = DATETIME_FORMAT) =>
+    value ? dayjs(value).tz(DANISH_TIME_ZONE_IDENTIFIER).format(format) : '';
+
+  /** Converts value of inner FormControl to type of outer FormControl. */
+  protected viewToModel = (value: string) => {
+    const date = dayjs(value, DATETIME_FORMAT, true);
+    if (!date.isValid()) return null;
+    return this.inclusive() ? date.endOf('m').toDate() : date.toDate();
+  };
+
+  // Must unfortunately be queried in order to update `activeDate`
+  private calendar = viewChild.required<MatCalendar<Date>>(MatCalendar);
+
+  // This inner FormControl is string only, but the outer FormControl is of type Date.
+  protected control = new FormControl('', { nonNullable: true });
+  protected selected = signal<Date | null>(null);
+  protected placeholder = computed(() => (this.locale.isDanish() ? DA_FILLER : EN_FILLER));
+  protected mask = maskitoDateTimeOptionsGenerator({
+    dateMode: 'dd/mm/yyyy',
+    timeMode: 'HH:MM',
+    dateSeparator: '-',
+  });
+
+  /** Set the label text for `watt-field`. */
+  label = input('');
+
+  /** When true, seconds will be set to 59 and milliseconds to 999. Otherwise, both are 0. */
+  inclusive = input(false);
+
+  /** Emits when the selected date has changed. */
+  dateChange = outputFromObservable(this.control.valueChanges.pipe(map(this.viewToModel)));
+
+  /** Emits when the field loses focus. */
+  blur = output<FocusEvent>();
+
+  protected handleBlur = (picker: HTMLElement, event: FocusEvent) => {
+    if (event.relatedTarget instanceof HTMLElement && picker.contains(event.relatedTarget)) {
+      const target = event.target as HTMLInputElement; // safe type assertion
+      setTimeout(() => target.focus());
+    } else {
+      picker.hidePopover();
+      this.blur.emit(event);
+    }
+  };
+
+  protected handleSelectedChange = (
+    field: HTMLInputElement,
+    picker: HTMLDivElement,
+    date: Date
+  ) => {
+    const prev = this.viewToModel(this.control.value);
+
+    // Only write the date part
+    field.value = prev
+      ? this.modelToView(dayjs(date).set('h', prev.getHours()).set('m', prev.getMinutes()).toDate())
+      : this.modelToView(date, PARTIAL_DATETIME_FORMAT);
+
+    field.dispatchEvent(new Event('input', { bubbles: true }));
+    picker.hidePopover();
+  };
+
+  constructor() {
+    this.dateChange.subscribe((date) => {
+      this.selected.set(date);
+      this.calendar().activeDate = date ?? new Date();
+    });
+  }
+
+  // Implementation for ControlValueAccessor
+  writeValue = (value: Date | null) => this.control.setValue(this.modelToView(value));
+  setDisabledState = (x: boolean) => (x ? this.control.disable() : this.control.enable());
+  registerOnTouched = (fn: () => void) => this.blur.subscribe(fn);
+  registerOnChange = (fn: (value: Date | null) => void) => this.dateChange.subscribe(fn);
+}
```

Comment:
praise: beautiful 🤩

Reply:
Appreciated, it wasn't easy cooking it down to that 😄